### DISCUSSION TITLE
My node version is 12+ , but it run a problem in gulp4

### DISCUSSION THREAD

**Discussion by xm-c** [gulpjs/gulp#2638] [Help]
#### This is the problem message

![image](https://user-images.githubusercontent.com/57081674/138838854-03b99b16-aaf1-46c6-90a5-3fb5d0bbd043.png)

And my gulp version and node version

![image](https://user-images.githubusercontent.com/57081674/138840232-abbd7a25-cca6-4939-b477-d01d801a2246.png)

![image](https://user-images.githubusercontent.com/57081674/138840457-e04a4673-6eda-4ebb-ae0e-bc4d22eeadf9.png)

**Comment by phated** [comment]
Our issue tracker is not a support forum. I'll transfer your issue to a Discussion, which is where you should be posting support questions

**Comment by robliou** [comment]
I have this problem as well. However, alongside gulp 4, I have tried using both Node 14.x and 16.x. and am also receiving the "primordials not defined" error.

The workaround bandied about on StackOverflow is to downgrade Node to version 11 or before, which I did attempt. However, I then receive another error ("globalThis is not defined"). This error apparently can only be resolved by _**upgrading**_ Node to a more recent version(!#!).

Hence, I am stuck between a rock and a hard place. What to do? How to solve this in order to get my app deployed? Has anyone been able to solve this...? Thanks,

**Reply by snerurkar-GM** [reply]
@xm-c @robliou any luck on finding the solution. We are also faced with the same issue.
